### PR TITLE
[ci/release] refactor retry function in release test

### PR DESCRIPTION
**Pull request by clarng**
Signed-off-by: Clarence Ng <clarence.wyng@gmail.com>

## Why are these changes needed?

No-op change to use the retry library, which will make it easier for us to customize the retry policy going forward.

We get intermittent infra failures that adds burden and slows us down when running nightly / testing PR

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

existing tests that test retry are still passing:

test_cluster_manager::testFindCreateClusterEnvFailFail
test_cluster_manager::testFindCreateClusterEnvFailSucceed
test_cluster_manager::testFindCreateClusterEnvSucceed

### CODE REVIEW

File: release/ray_release/tests/test_retry.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import sys
```
Comment (krfricke) [inline release/ray_release/tests/test_retry.py:1]: We don't really need this test, right? Assuming that the retry library just works

**Review by krfricke — APPROVED** [review]
Generally looks good, just quick question about the test

**Comment by krfricke** [comment]
The release tests seems to be failing here: 

```
ModuleNotFoundError: No module named 'retry'
```

May need to install them for CI (requirements_buildkite.txt)

File: release/ray_release/tests/test_retry.py (hunk 1 above)
Comment (clarng) [reply release/ray_release/tests/test_retry.py]: done

Comment (clarng) [reply release/ray_release/tests/test_retry.py]: done

Comment (clarng) [reply release/ray_release/tests/test_retry.py]: done

**Review by rkooo567 — APPROVED** [review]
CAn you make sure it works as expected before merging it?

**Comment by krfricke** [comment]
Tests are passing and release tests still succeed, but I can't easily create a failing condition on staging. I'm merging since it doesn't seem to harm the current pipeline.

**Pull request merged**